### PR TITLE
PA-597 Java standalone client changes

### PR DESCRIPTION
**Pull request by sergvatu**
This PR includes: 
1. the vector module was added to run related tests with enterprise server
2. I added skipping redundant run of OS tests against enterprise sever since we run them agains OS server and they should work agains enterprise server by default

### CODE REVIEW

File: .github/workflows/java_client_compatibility.yaml [hunk 1]
```diff
@@ -81,13 +81,29 @@ jobs:
...
 
+      - name: Build modules
+        shell: bash -l {0}
+        run: |
+          chmod +x mvnw
+          ./mvnw -B -V -e clean install -DskipTests
+        working-directory: client
+
       - name: Run non-enterprise tests
+        if: ${{ matrix.kind == 'os' }}
```
Comment (ihsandemir) [inline .github/workflows/java_client_compatibility.yaml:92]: you should still run the OS related tests(e.g. IMap tests) against the EE server.

File: .github/workflows/server_compatibility.yaml [hunk 2]
```diff
@@ -909,12 +909,29 @@ jobs:
...
 
+      - name: Build modules
+        shell: bash -l {0}
+        run: |
+          chmod +x mvnw
+          ./mvnw -B -V -e clean install -DskipTests
+        working-directory: client
+
       - name: Run non-enterprise tests
+        if: ${{ matrix.server_kind == 'os' }}
```
Comment (ihsandemir) [inline .github/workflows/server_compatibility.yaml:920]: os scoped tests also should run against ee server.

File: .github/workflows/java_client_compatibility.yaml (hunk 1 above)
Comment (sergvatu) [reply .github/workflows/java_client_compatibility.yaml]: I added them as a separate job

File: .github/workflows/server_compatibility.yaml (hunk 2 above)
Comment (sergvatu) [reply .github/workflows/server_compatibility.yaml]: I added them as a separate job

File: .github/workflows/server_compatibility.yaml [hunk 3]
```diff
@@ -808,8 +808,11 @@ jobs:
       matrix:
         client_tag: ${{ fromJson(needs.setup_java_client_matrix.outputs.matrix) }}
         server_kind: [ os, enterprise ]
-
-    name: Test Java Thin client ${{ matrix.client_tag }} with ${{ matrix.server_kind }} server
+        tests_type: [ os, enterprise ]
+        exclude:
+          - server_kind: os
+            tests_type: enterprise
+    name: Test Java client ${{ matrix.client_tag }} branch running ${{ matrix.tests_type }} tests against ${{ matrix.server_kind }} server
```
Comment (ihsandemir) [inline .github/workflows/server_compatibility.yaml:815]:
```suggestion
    name: Test Java client (standalone) ${{ matrix.client_tag }} branch running ${{ matrix.tests_type }} tests against ${{ matrix.server_kind }} server
```

**Review by ihsandemir — COMMENTED** [review]
Looks good but can we have a sample run for both client and server compat. to make sure the changes work?

**Review by ihsandemir — APPROVED** [review]

**Pull request merged**